hare/algorithm.go

```diff
@@ -255,39 +256,44 @@ func (proc *ConsensusProcess) onEarlyMessage(m *Msg) {
 		return
 	}
 
-	if m.HareMessage == nil {
-		proc.Error("onEarlyMessage called with nil message")
+	if m.XDRMessage == nil {
+		proc.Error("onEarlyMessage called with nil Message")
+		return
+	}
+
+	if m.Message == nil {
+		proc.Error("onEarlyMessage called with nil inner message")
 		return
 	}
 
 	pub := signing.NewPublicKey(m.PubKey)
 	if _, exist := proc.pending[pub.String()]; exist { // ignore, already received
-		proc.Warning("Already received message from sender %v", pub.String())
+		proc.Warning("Already received Message from sender %v", pub.String())
 		return
 	}
 
 	proc.pending[pub.String()] = m
 }
 
 func (proc *ConsensusProcess) handleMessage(m *Msg) {
-	// Note: instanceId is already verified by the broker
+	// Note: InstanceId is already verified by the broker
 
-	proc.Debug("Received message %v", m)
+	proc.Debug("Received Message %v", m)
 
 	if !proc.validator.SyntacticallyValidateMessage(m) {
 		proc.Warning("Syntactically validation failed, pubkey %v", m.PubKey)
 		return
 	}
 
 	mType := MessageType(m.Message.Type).String()
-	// validate message for this or next round
+	// validate Message for this or next round
 	if !proc.validator.ContextuallyValidateMessage(m, proc.k) {
 		if !proc.validator.ContextuallyValidateMessage(m, proc.k+1) {
-			// TODO: should return error from message validation to indicate what failed, should retry only for contextual failure
+			// TODO: should return error from Message validation to indicate what failed, should retry only for contextual failure
 			proc.Warning("Message of type %v is not valid for either round, pubkey %v", mType, m.PubKey)
 			return
-		} else { // a valid early message, keep it for later
-			proc.Info("Early message of type %v detected. Keeping message, pubkey %v", mType, m.PubKey)
+		} else { // a valid early Message, keep it for later
+			proc.Info("Early Message of type %v detected. Keeping Message, pubkey %v", mType, m.PubKey)
```

Comment:
use bytes to hex on pubkey, show only 5 characters from pubkey

Reply:
Fixed